src/phoenix/server/api/auth.py

```diff
@@ -42,3 +42,12 @@ def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
         if not info.context.auth_enabled:
             return False
         return isinstance((user := info.context.user), PhoenixUser) and user.is_admin
+
+
+class IsAdminIfAuthEnabled(Authorization):
+    message = MSG_ADMIN_ONLY
+
+    def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
+        if not info.context.auth_enabled:
+            return True
+        return isinstance((user := info.context.user), PhoenixUser) and user.is_admin
```

Comment:
Would it work to split the existing `IsAdmin` into two permissions?


```suggestion
class RequiresAuth(Authorization):
    message = "add message here"

    def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
        if not info.context.auth_enabled:
            return False
        return True

class IsAdmin(Authorization):
    message = MSG_ADMIN_ONLY

    def has_permission(self, source: Any, info: Info, **kwargs: Any) -> bool:
        return isinstance((user := info.context.user), PhoenixUser) and user.is_admin
```